webui/src/App.js

```diff
@@ -575,17 +655,58 @@ function App() {
           </Navbar>
           <Switch>
             <Route exact path="/">
-              <WebUI />
+              <ValueMonitors emit={emit} numPanels={numPanels} />
             </Route>
             <Route path="/plot">
               <Plot />
             </Route>
           </Switch>
         </Router>
       </div>
-    :
-    <></>
+    </CurValuesRefContext.Provider>
+  );
+}
+
+function LoadingScreen() {
+  return (
+    <div style={{ color: "white", height: "100vh", width: "100vw" }}>
+      <Navbar bg="light">
+        <Navbar.Brand as={"span"} to="/">FSR WebUI</Navbar.Brand>
+      </Navbar>
+      <div style={{
+        backgroundColor: "#282c34",
+        border: "1px solid white",
+        fontSize: "1.25rem",
+        padding: "0.5rem 1rem",
+        height: "96vh"
+      }}>
+        Connecting...
+      </div>
+    </div>
   );
 }
 
+function App() {
+  const { defaults, reloadDefaults } = useDefaults();
+  const {
+    curValuesRef,
+    emit,
+    isWsReady,
+    wsCallbacksRef
+  } = useWsConnection({ defaults, onCloseWs: reloadDefaults });
+
+  if (defaults && isWsReady) {
+    return (
+      <FSRWebUI
+        curValuesRef={curValuesRef}
+        defaults={defaults}
+        emit={emit}
+        wsCallbacksRef={wsCallbacksRef}
+      />
+    );
+  } else {
+    return <LoadingScreen />
```

Comment:
I want to note that if the websocket closes, most of the UI is unmounted and historical sensor values are cleared, which is a change to existing behavior.

Reply:
I thinnnnnnk this is fine. I can't really think of a scenario where it'd be particularly breaking.

Reply:
I think this is probably fine as well. Can't think of a huge downside do it.